pkg/snowflake/parameters.go

```diff
@@ -683,52 +683,119 @@ func GetParameterDefault(key string) ParameterDefault {
 	return ParameterDefaults()[key]
 }
 
-// ParameterBuilder abstracts the creation of SQL queries for Snowflake parameters.
-type ParameterBuilder struct {
+// AccountParameterBuilder abstracts the creation of SQL queries for Snowflake account parameters.
+type AccountParameterBuilder struct {
+	key   string
+	value string
+	db    *sql.DB
+}
+
+func NewAccountParameter(key, value string, db *sql.DB) *AccountParameterBuilder {
+	return &AccountParameterBuilder{
+		key:   key,
+		value: value,
+		db:    db,
+	}
+}
+
+func (v *AccountParameterBuilder) SetParameter() error {
+	stmt := fmt.Sprintf("ALTER ACCOUNT SET %s = %s", v.key, v.value)
+	_, err := v.db.Exec(stmt)
+	return err
+}
+
+// SessionParameterBuilder abstracts the creation of SQL queries for Snowflake session parameters.
+type SessionParameterBuilder struct {
+	key       string
+	value     string
+	onAccount bool
+	user      string
+	db        *sql.DB
+}
+
+func NewSessionParameter(key, value string, db *sql.DB) *SessionParameterBuilder {
+	return &SessionParameterBuilder{
+		key:   key,
+		value: value,
+		db:    db,
+	}
+}
+
+func (v *SessionParameterBuilder) SetOnAccount(onAccount bool) *SessionParameterBuilder {
+	v.onAccount = onAccount
+	return v
+}
+
+func (v *SessionParameterBuilder) SetUser(user string) *SessionParameterBuilder {
+	v.user = user
+	return v
+}
+
+func (v *SessionParameterBuilder) SetParameter() error {
+	if v.onAccount {
+		// prepared statements do not work here for some reason. We already validate inputs so its okay
+		stmt := fmt.Sprintf("ALTER ACCOUNT SET %s = %s", v.key, v.value)
+		_, err := v.db.Exec(stmt)
+		return err
```

Comment:
Could we reuse `AccountParameterBuilder.SetParameter` to avoid duplicating this SQL? I assume these two use cases are functionally identical.